recipes-sota/aktualizr/aktualizr_git.bb

```diff
@@ -32,29 +27,34 @@ BRANCH ?= "master"
 
 S = "${WORKDIR}/git"
 
-inherit cmake
-
-inherit systemd
+inherit pkgconfig cmake systemd
 
 SYSTEMD_PACKAGES = "${PN} ${PN}-secondary"
 SYSTEMD_SERVICE_${PN} = "aktualizr.service"
 SYSTEMD_SERVICE_${PN}-secondary = "aktualizr-secondary.socket"
 
-BBCLASSEXTEND =+ "native"
-
-require garage-sign-version.inc
-
-EXTRA_OECMAKE = "-DWARNING_AS_ERROR=OFF \
-                 -DCMAKE_BUILD_TYPE=Release \
-                 -DAKTUALIZR_VERSION=${PV} \
-                 -DBUILD_LOAD_TESTS=OFF"
-EXTRA_OECMAKE_append_class-target = " -DBUILD_OSTREE=ON \
-                                      ${@bb.utils.contains('SOTA_CLIENT_FEATURES', 'hsm', '-DBUILD_P11=ON', '', d)} "
-EXTRA_OECMAKE_append_class-native = " -DBUILD_SOTA_TOOLS=ON \
-                                      -DBUILD_OSTREE=OFF \
-                                      -DBUILD_SYSTEMD=OFF \
-                                      -DGARAGE_SIGN_VERSION=${GARAGE_SIGN_VERSION} \
-                                      -DGARAGE_SIGN_SHA256=${GARAGE_SIGN_SHA256}"
+EXTRA_OECMAKE = "-DCMAKE_BUILD_TYPE=Release -DAKTUALIZR_VERSION=${PV}"
+
+GARAGE_SIGN_OPS = "${@ '-DGARAGE_SIGN_VERSION=%s' % d.getVar('GARAGE_SIGN_VERSION') if d.getVar('GARAGE_SIGN_VERSION') is not None else ''} \
+                   ${@ '-DGARAGE_SIGN_SHA256=%s' % d.getVar('GARAGE_SIGN_SHA256') if d.getVar('GARAGE_SIGN_SHA256') is not None else ''} \
+                  "
+
+PACKAGECONFIG ?= "ostree ${@bb.utils.filter('DISTRO_FEATURES', 'systemd', d)} ${@bb.utils.filter('SOTA_CLIENT_FEATURES', 'hsm serialcan ubootenv', d)}"
```

Comment:
Using bb.utils.filter for `SOTA_CLIENT_FEATURES` is a nice touch here, I like that!